packages/ai/src/ai/account/file_store.py

```
@@ -350,6 +350,72 @@ async def mkdir(self, path: str) -> None:
+            # Directory rename: copy every file to the new prefix then delete
+            new_dir_prefix = new_full.rstrip('/') + '/'
+            for file_path in all_files:
+                relative_to_old = file_path[len(dir_prefix) :]
+                new_file_path = new_dir_prefix + relative_to_old
+                data = await self._store.read_bytes(file_path)
+                await self._store.write_bytes(new_file_path, data)
+                await self._store.delete_file(file_path)
+        else:
+            # File rename: read + write new + delete old
+            if old_full in self._write_locks:
+                raise StorageError(f'Cannot rename file while it is open for writing: {old_path}')
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)